pandas/tests/series/methods/test_convert_dtypes.py

```diff
@@ -225,3 +225,14 @@ def test_convert_bool_dtype(self):
         # GH32287
         df = pd.DataFrame({"A": pd.array([True])})
         tm.assert_frame_equal(df, df.convert_dtypes())
+
+    def test_convert_object_to_datetime_with_cache(self):
+        # GH#39882
+        ser = pd.Series(
+            [None] + [pd.NaT] * 50 + [pd.Timestamp("2012-07-26")], dtype="object"
```

Comment:
could make 50 parameter in tools.datetimes, ensures that the test stays valid even if that gets changed

Reply:
Sounds good. Like this?